api/v1alpha1/repository_types.go

```diff
@@ -32,6 +32,19 @@ type PullStategy struct {
 	Retry int `json:"retry,omitempty"`
 }
 
+type FilterCond struct {
+	// If True, the current version will be retained even if it is deprecated.
+	Deprecated bool `json:"deprecated,omitempty"`
+	// Accurately match each item in the versions
+	Versions []string `json:"versions,omitempty"`
+	// Filter version by regexp
+	Regexp string `json:"regexp,omitempty"`
+
+	// All versions smaller/greater than a particular version
+	Than string `json:"than,omitempty"`
```

Comment:
add a comment to point this constraint to `https://github.com/Masterminds/semver#semver`